### PR TITLE
Add reproducibility index file

### LINKED ISSUE #521: Add index page for reproducibility section
This `index.md` file has been created but currently has no content. Eventually, it should contain an overview of how we ensure reproducibility throughout the OpenScPCA project. This page will include the reproducibility overview figure (#491). 

We should also include a section about how you should always be working from the `openscpca` conda environment (or your module's conda environment, as the case may be) when developing.

### PR DESCRIPTION
**Pull request by sjspielman**
Closes #521

This PR adds an index page to the reproducibility section. Please let me know how the scope and level of detail looks here! I also thought about maybe adding a larger paragraph at the top talking more generally about reproducibility, but I thought that would detract too much from the main points of how we maintain reproducibility in OpenScPCA specifically.

I also moved the software environment docs up in the navbar to better match the order of the figure, which I added, maybe temporarily, just to see how it looks! (@dvenprasad, I think we need bigger font?)

### CODE REVIEW

File: mkdocs.yml [hunk 1]
```diff
@@ -133,15 +133,15 @@ nav:
...
-    - Automated OpenScPCA workflows:
-      - ensuring-repro/workflows/index.md
-      - ensuring-repro/workflows/run-module-gha.md
-      - ensuring-repro/workflows/build-docker-gha.md
     - Managing software environments:
       - ensuring-repro/managing-software/index.md
       - ensuring-repro/managing-software/using-renv.md
       - ensuring-repro/managing-software/using-conda.md
       - ensuring-repro/managing-software/reporting-dependencies.md
+    - Automated OpenScPCA workflows:
```
Comment (jashapiro) [inline mkdocs.yml:141]: This reorder makes sense to me.

File: docs/ensuring-repro/index.md [hunk 2]
```diff
@@ -1,3 +1,15 @@
...
+
+<figure markdown="span">
+    ![Overview of the OpenScPCA reproducibility framework.](../img/reproducibility-overview.png){width="700"}
+</figure>
+
+- We require each analysis module to have [its own software environment](./managing-software/index.md) with pinned dependencies
+    - As analysis modules mature, these software dependencies will be added to a [module-specific Docker image](./docker/docker-images.md)
+- We use the [GitHub Actions](https://docs.github.com/en/actions) continuous integration framework to [automate certain processes that ensure reproducibility](./workflows/index.md), including:
+    - [Testing](./workflows/run-module-gha.md) that analysis modules run to completion without errors
+    - [Building and pushing module-specific Docker images](./workflows/build-docker-gha.md) to a public registry for broad reuse
```
Comment (jashapiro) [inline docs/ensuring-repro/index.md:14]: Small rearrangment
```suggestion
    - [Building module-specific Docker images](./workflows/build-docker-gha.md) and pushing them to a public registry for broad reuse
```

File: docs/ensuring-repro/index.md [hunk 3]
```diff
@@ -1,3 +1,15 @@
...
+- We maintain a separate repository [`OpenScPCA-nf`](./openscpca-nf/index.md), where completed analysis modules are run in a reproducible [Nextflow workflow](https://www.nextflow.io/) to generate workflow results associated with OpenScPCA data releases
```
Comment (jashapiro) [inline docs/ensuring-repro/index.md:15]:
```suggestion
- We maintain a separate repository, [`OpenScPCA-nf`](./openscpca-nf/index.md), where completed analysis modules are run in a reproducible [Nextflow workflow](https://www.nextflow.io/) to generate workflow results associated with OpenScPCA data releases
```

**Review by jashapiro — COMMENTED** [review]
LGTM, with some minor comments

**Review by jashapiro — APPROVED** [review]
Sorry, I had meant to approve last time!

**Review by dvenprasad — APPROVED** [review]
🐌 🐌 🐌

**Pull request merged**